cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -700,8 +779,74 @@ predicate addressFlowInstrRTC(Instruction iFrom, Instruction iTo) {
   or
   AddressFlow::addressFlowInstrTC(iFrom, iTo)
 }
-  // Instruction -> Operand flow
-  simpleOperandLocalFlowStep(nodeFrom.asInstruction(), nodeTo.asOperand())
+
+/**
+ * Holds if `instr2` is a possible (transitive) successor of `instr1` in the control-flow graph.
+ *
+ * This predicate is `inline` because it's infeasible to compute it in isolation.
+ */
+pragma[inline]
+private predicate isSuccessorInstruction(Instruction instr1, Instruction instr2) {
+  exists(IRBlock block1, IRBlock block2 |
+    block1 = instr1.getBlock() and
+    block2 = instr2.getBlock() and
+    if block1 = block2
+    then
+      exists(int index1, int index2 |
+        block1.getInstruction(index1) = instr1 and
+        block2.getInstruction(index2) = instr2 and
+        index1 < index2
+      )
+    else block1.getASuccessor+() = block2
+  )
+}
+
+/**
+ * Holds if `nodeTo` should receive flow after leaving `nodeFrom`. This occurs in two different
+ * situations:
+ *   - There is a subsequence of use a memory operand that has been used when traversing the chain of
+ *     addresses used to perform a sequence of `storeStep`s, or
```

Comment:
I think there's a copy-paste mistake here. Is this just the steps up the address computation chain?

Reply:
It's not a copy-paste mistake per se. But I understand your confusion. The comment wasn't very clear. 788b5e012e654a7f50af07b560b7529fd59b3f37 adds more explanation (and an example) to the predicate now. Hopefully, that makes it more clear.